tests/test_qaoa.py

```diff
@@ -110,27 +117,228 @@ def matrix(hamiltonian: qml.Hamiltonian, n_wires: int) -> csc_matrix:
     ops_matrices = []
 
     for op in hamiltonian.ops:
-        op_wires = np.array(op.wires.tolist())
-        op_list = op.non_identity_obs if isinstance(op, qml.operation.Tensor) else [op]
-        op_matrices = []
 
-        for wire in range(n_wires):
-            loc = np.argwhere(op_wires == wire).flatten()
-            mat = np.eye(2) if len(loc) == 0 else op_list[loc[0]].matrix()
-            mat = csc_matrix(mat)
-            op_matrices.append(mat)
+        if isinstance(op, qml.ops.Prod):
+            op_matrix = op.sparse_matrix(wire_order=list(range(n_wires)))
+        else:
+            op_wires = np.array(op.wires.tolist())
+            op_list = op.non_identity_obs if isinstance(op, qml.operation.Tensor) else [op]
+            op_matrices = []
 
-        op_matrix = op_matrices.pop(0)
+            for wire in range(n_wires):
+                loc = np.argwhere(op_wires == wire).flatten()
+                mat = np.eye(2) if len(loc) == 0 else op_list[loc[0]].matrix()
+                mat = csc_matrix(mat)
+                op_matrices.append(mat)
 
-        for mat in op_matrices:
-            op_matrix = kron(op_matrix, mat)
+            op_matrix = op_matrices.pop(0)
+
+            for mat in op_matrices:
+                op_matrix = kron(op_matrix, mat)
 
         ops_matrices.append(op_matrix)
 
     mat = sum(coeff * op_mat for coeff, op_mat in zip(hamiltonian.coeffs, ops_matrices))
     return csc_matrix(mat)
 
 
+def make_xy_mixer_test_cases():
```

Comment:
> You can also leave the original parametrization and `convert_to_legacy_H`

That was the original approach but it caused many issues. The problem was that the Hamiltonians created in the legacy op math context and the Hamiltonians created with new op math enabled and converted to legacy are equivalent but not the same, which is causing many tests to fail.

Reply:
Let me give it a try.

Reply:
Actually `convert_to_legacy_H` seems to be working now. I just made an update. I had to leave `TestLayers` as is because legacy opmath and new opmath has inconsistent ordering of wires for `PauliRot`.

Reply:
nice!

Reply:
The inconsistent wire ordering is likely due to the pauli rep 😢

Reply:
> The inconsistent wire ordering is likely due to the pauli rep 😢

Yeah it's not a big deal since they are obviously equivalent, we just don't have a proper function to check for that equivalence. I believe this is already on the technical backlog as discussed in a meeting last week.

Reply:
yep :)